src/server/api.js

```diff
@@ -0,0 +1,372 @@
+
+const getTimestampIntervals = (numberOfDays, meta) => {
+  let timestampIntervals = [];
+  for (let dayIndex = 0; dayIndex < numberOfDays; dayIndex++) {
+    timestampIntervals.push({
+      start: getStartTimestampForDay(dayIndex, meta),
+      end: getEndTimestampForDay(dayIndex, meta)
+    });
+  }
+  return timestampIntervals;
+};
+
+const getDatesTimesAndPrices = (
+  close,
+  gmtoffset,
+  numberOfDays,
+  timestamp,
+  timestampIntervals
+) => {
+  let datesTimesAndPrices = [];
+  for (let dayIndex = 0; dayIndex < numberOfDays; dayIndex++) {
+    const intervalOfDatesTimesAndPrices = getDatesAndTimesForOneDay(
+      close,
+      dayIndex,
+      gmtoffset,
+      numberOfDays,
+      timestamp,
+      timestampIntervals
+    );
+    datesTimesAndPrices = datesTimesAndPrices.concat(
+      intervalOfDatesTimesAndPrices
+    );
+  }
+  return datesTimesAndPrices;
+};
+
+// numberOfDays much match the range used to obtain the intradayRes.
+const getIntradayStockData = (intradayRes, numberOfDays) => {
+  const intradayData = JSON.parse(intradayRes);
+  const result = intradayData.chart.result[0];
+  const {
+    indicators,
+    meta,
+    timestamp
+  } = result;
+  const { gmtoffset } = meta;
+
+  // Array of objects, one for each day.
+  // Each contains a start timestamp and end timestamp for that trading day.
+  const timestampIntervals = getTimestampIntervals(numberOfDays, meta);
+
+  const { close } = indicators.quote[0];
+  const datesTimesAndPrices = getDatesTimesAndPrices(
+    close,
+    gmtoffset,
+    numberOfDays,
+    timestamp,
+    timestampIntervals
+  );
+  return datesTimesAndPrices;
+};
+
+const getQueryForIntradayData = (symbol, range, interval) => {
+  return `https://query1.finance.yahoo.com/v8/finance/chart/${symbol}` +
+  `?range=${range}&includePrePost=true&interval=${interval}` +
+  '&corsDomain=finance.yahoo.com&.tsrc=finance';
+};
+
+const requestMaxStockData = (symbol, callback) => {
+  historical({
+    // gets all data because we didn't specify from/to
+    symbol,
+    period: 'd'
+  }).then(
+    dailyData => {
+      console.log('max');
+      if (!dailyData[0]) {
+        throw new Error('Historical data was not found.');
+      }
+      callback(null, getDatesAndPrices(dailyData));
+    });
+};
+
+const requestOneDayStockData = (symbol, callback) => {
+  const queryOneDay = getQueryForIntradayData(
+    symbol,
+    QUERY_RANGE_ONE_DAY,
+    QUERY_INTERVAL_ONE_DAY
+  );
+  rp(queryOneDay)
+    .then(oneDayRes => {
+      console.log('one');
+      callback(null, getIntradayStockData(oneDayRes, ONE_DAY));
+    });
+};
+
+const requestFiveDayStockData = (symbol, callback) => {
+  const queryFiveDay = getQueryForIntradayData(symbol,
+    QUERY_RANGE_FIVE_DAY,
+    QUERY_INTERVAL_FIVE_DAY
+  );
+  rp(queryFiveDay)
+    .then(fiveDayRes => {
+      console.log('five');
+      callback(null, getIntradayStockData(fiveDayRes, FIVE_DAYS));
+    });
+};
+
+router.get('/stocks/:symbol', (req, res) => {
```

Comment:
I want to move a bunch of the helper functions to another file so that this one can be more concerned with just routing.  Keep the `router.get` and maybe the functions it directly calls, and move the rest elsewhere